### PR TITLE
Fixes restart tracker's incorrect subscription #26 #27

### PR DESCRIPTION
**Pull request by matevarga**

### CODE REVIEW

File: src/test/java/com/pkb/unit/tracker/TrackerRestartTest.java [hunk 1]
```diff
@@ -68,4 +71,42 @@ public void restartTrackerDoesntEmitWhenUnitFailsToRestart() {
...
+
+        // WHEN
+        TestObserver<Boolean> testRestartedObserver = Tracker.unitRestarted(bus, unit1id).test();
+        unit2.stop();
+        unit2.completeStop();
+        unit1.completeStop();
+        unit2.start();
+        unit1.failStart();
+        testScheduler.triggerActions();
+
```
Comment (MFAshby) [inline src/test/java/com/pkb/unit/tracker/TrackerRestartTest.java:106]: You probably also want to assert the system state here too; unit2 should have successfully started (proving that a unit started, but the restartedObserver didn't trigger)

**Review by MFAshby — CHANGES_REQUESTED** [review]
I think one additional assertion should be made in the test; to verify the system state as well as verifying that the restartedTracker didn't fire (in case there was a bug with the test and no units actually restarted) 

Otherwise LGTM (my bad for introducing this bug :) )

File: src/test/java/com/pkb/unit/tracker/TrackerRestartTest.java (hunk 1 above)
Comment (matevarga) [reply src/test/java/com/pkb/unit/tracker/TrackerRestartTest.java:106]: ok, fixing

**Review by MFAshby — APPROVED** [review]
LGTM

**Review by ownimage — APPROVED** [review]
I haven't been into the internals of this but it does fix the problem reported.

**Pull request merged**